esx_service/utils/vmdk_utils.py

```diff
@@ -109,21 +122,21 @@ def get_datastores():
     'url' is datastore URL (e.g. '/vmfs/volumes/vsan:572904f8c031435f-3513e0db551fcc82')
     'dockvol-path; is a full path to 'dockvols' folder on datastore
     """
-    if datastores == None:
-        init_datastoreCache()
+    init_datastoreCache()
     return datastores
```

Comment:
Another thread may in parallel be in init_datastoreCache() and set datastores to None (by force option). May be get_datastores() should be returning a copy of the datastores list.